Add logo provider and tokens to theme-shared package

fahrigedik opened the pull request:
### Description

Resolves #23113 (write the related issue number if available)

TODO: Add logo provider and tokens to theme-shared package

### Checklist

- [ ] I fully tested it as developer / designer and created unit / integration tests
- [ ] I documented it (or no need to document or I will create a separate documentation issue)

### How to test it?

you have to add on `npm\ng-packs\apps\dev-app\src\app\app.config.ts`
```
//other imports
import { provideLogo, withEnvironmentOptions } from '@abp/ng.theme.shared';
//
//other providers..
//
provideLogo(withEnvironmentOptions(environment)),
```

and
you have to add `logoUrl` on `npm\ng-packs\apps\dev-app\src\environments\environment.ts` file.

sample environment.ts 
```  application: {
    baseUrl,
    name: 'MyProjectNamee',
    logoUrl: 'https://cdn-icons-png.flaticon.com/512/9908/9908191.png',
  },```
  
  and you have to run dev-app

fahrigedik committed:
Add logo provider and tokens to theme-shared package

sumeyyeKurtulus commented on npm/ng-packs/packages/theme-shared/src/lib/providers/logo.provider.ts:
```diff
@@ -0,0 +1,51 @@
+} from '@angular/core';
+import { LOGO_APP_NAME_TOKEN, LOGO_URL_TOKEN } from '../tokens/logo.token';
+
+export enum LpxLogoFeatureKind {
```
The `lpx` prefix stands for the lepton-x theme. That is why, we can make it more generalized since all themes are tended to use the same provider.

sumeyyeKurtulus reviewed (commented):
Hello @fahrigedik thank you for putting an effort to migrate this provider here.
 
It seems mergable only if we can make the theme basic use the tokens and environment variables as lepton-x theme does `npm/ng-packs/packages/theme-basic/src/lib/components/logo/logo.component.ts`

fahrigedik committed:
Rename LpxLogoFeature to LogoFeature in provider

sumeyyeKurtulus reviewed (commented):
> It seems mergable only if we can make the theme basic use the tokens and environment variables as lepton-x theme does `npm/ng-packs/packages/theme-basic/src/lib/components/logo/logo.component.ts`

It is good to go once we also make the basic theme use the same provider approach here.

fahrigedik committed:
Refactor LogoComponent to use injectable tokens

Updated LogoComponent to use LOGO_URL_TOKEN and LOGO_APP_NAME_TOKEN for logo URL and app name, falling back to environment values if tokens are not provided. This improves flexibility and allows for easier customization of the logo and app name.

fahrigedik committed:
Merge branch 'dev' into issue-23113
